chef/cookbooks/provisioner/recipes/bootdisk.rb

```diff
@@ -78,17 +79,36 @@
         node[:crowbar_wall][:boot_device] = "disk/by-id/#{bootdisk}"
       end
     end
+
+    # Searching boot disk with virtio driver if no other disks were found
+    # on SLES_12 sleshammer image
+    if dev.nil?
+      basedir = "/sys/block"
+      ::Dir.entries(basedir).sort.each do |path|
+        next unless File.symlink?(File.join(basedir, path))
+        dev = File.readlink("#{basedir}/#{path}").split("/")[-1]
+        if dev =~ /^vd[a-z]+$/
```

Comment:
Ok, new proposal would be 
```
      basedir = "/sys/block"
      ::Dir.entries(basedir).sort.each do |path|
        next unless File.symlink?(File.join(basedir, path))
        dev = File.readlink("#{basedir}/#{path}")
        if dev =~ /virtio/
          dev = dev.split("/")[-1]
          node[:crowbar_wall][:boot_device] = dev
          break
        end
      end
```
This part was moved earlier before `raise` condition